joblib/test/test_memory_async.py

```diff
@@ -0,0 +1,150 @@
+import asyncio
+import gc
+import shutil
+
+import pytest
+
+from joblib.memory import (AsyncMemorizedFunc, AsyncNotMemorizedFunc,
+                           MemorizedResult, Memory, NotMemorizedResult)
+from joblib.test.common import np, with_numpy
+from joblib.testing import raises
+
+from .test_memory import (corrupt_single_cache_item,
+                          monkeypatch_cached_func_warn)
+
+
+async def f(x, y=1):
+    await asyncio.sleep(0.1)
+    return x ** 2 + y
```

Comment:
This function is only used in `test_call_and_shelve_async`, so probably better to define it in the test?